tools/junitreport/pkg/api/junit.xsd

```diff
@@ -0,0 +1,203 @@
+            <xs:simpleType>
+               <xs:restriction base="pre-string">
+                  <xs:whiteSpace value="preserve" />
+               </xs:restriction>
+            </xs:simpleType>
+         </xs:element>
+      </xs:sequence>
+      <xs:attribute name="name" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">Full class name of the test for non-aggregated testsuite documents. Class name without the package for aggregated testsuites documents</xs:documentation>
+         </xs:annotation>
+         <xs:simpleType>
+            <xs:restriction base="xs:token">
+               <xs:minLength value="1" />
+            </xs:restriction>
+         </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="timestamp" type="ISO8601_DATETIME_PATTERN" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">when the test was executed. Timezone may not be specified.</xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="hostname" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">Host on which the tests were executed. 'localhost' should be used if the hostname cannot be determined.</xs:documentation>
+         </xs:annotation>
+         <xs:simpleType>
+            <xs:restriction base="xs:token">
+               <xs:minLength value="1" />
+            </xs:restriction>
+         </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="tests" type="xs:int" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">The total number of tests in the suite</xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="failures" type="xs:int" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">The total number of tests in the suite that failed. A failure is a test which the code has explicitly failed by using the mechanisms for that purpose. e.g., via an assertEquals</xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="errors" type="xs:int" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">The total number of tests in the suite that errorrd. An errored test is one that had an unanticipated problem. e.g., an unchecked throwable; or a problem with the implementation of the test.</xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="time" type="xs:decimal" use="required">
+         <xs:annotation>
+            <xs:documentation xml:lang="en">Time taken (in seconds) to execute the tests in the suite</xs:documentation>
+         </xs:annotation>
+      </xs:attribute>
+   </xs:complexType>
+   <xs:simpleType name="pre-string">
+      <xs:restriction base="xs:string">
+         <xs:whiteSpace value="preserve" />
+      </xs:restriction>
+   </xs:simpleType>
+</xs:schema>
```

Comment:
I guess I could refactor this to YAML as it's only used as a source of truth, not for processing, etc.